build-scripts/build_tests.py

```diff
@@ -0,0 +1,244 @@
+
+
+def _process_shared_file(env_yaml: dict, file: pathlib.Path, shared_output_path: pathlib.Path) \
+        -> None:
+    file_contents = ssg.jinja.process_file_with_macros(str(file.absolute()), env_yaml)
+    shared_script_path = shared_output_path / file.name
+    _write_path(file_contents, shared_script_path)
+
+
+def _copy_and_process_shared(env_yaml: dict, output_path: pathlib.Path, root_path: pathlib.Path) \
+        -> None:
+    tests_shared_root = root_path / "tests" / "shared"
+    shared_output_path = output_path / "shared"
+    shared_output_path.mkdir(parents=True, exist_ok=True)
+    for file in tests_shared_root.iterdir():  # type: pathlib.Path
+        # We only support one level deep, this avoids recursive functions
+        if file.is_dir():
+            for sub_file in file.iterdir():
+                shared_output_path_sub = shared_output_path / file.name
+                shared_output_path_sub.mkdir(parents=True, exist_ok=True)
+                _process_shared_file(env_yaml, sub_file, shared_output_path_sub)
+        else:
+            _process_shared_file(env_yaml, file, shared_output_path)
+
+
+def _get_platform_from_file_contents(file_contents: str) -> str:
+    # Some tests don't have an explict platform assume always applicable
+    platform = "multi_platform_all"
+    for line in file_contents.split("\n"):
+        if not line.startswith('#'):
+            # The loop is now in the main test content, stop processing the file.
+            break
+        if line.startswith('# platform'):
+            platform_parts = line.split('=')
+            if len(platform_parts) == 2:
+                platform = platform_parts[1]
+                break
+    return platform.strip()
+
+
+def _process_local_tests(product: str, env_yaml: dict, rule_output_path: pathlib.Path,
+                         rule_tests_root: pathlib.Path) -> None:
+    logger = logging.getLogger()
+    for test in rule_tests_root.iterdir():  # type: pathlib.Path
+        if test.is_dir():
+            logger.warning("Skipping directory %s in rule %s", test.name,
+                           rule_output_path.name)
+            continue
+        if not _is_test_file(test.name):
+            file_contents = ssg.jinja.process_file_with_macros(str(test.absolute()),
+                                                               env_yaml)
+            output_file = rule_output_path / test.name
+            rule_output_path.mkdir(parents=True, exist_ok=True)
+            _write_path(file_contents, output_file)
+        file_contents = test.read_text()
+        platform = _get_platform_from_file_contents(file_contents)
+        if ssg.utils.is_applicable_for_product(platform, product):
+            content = ssg.jinja.process_file_with_macros(str(test.absolute()), env_yaml)
+            rule_output_path.mkdir(parents=True, exist_ok=True)
+            _write_path(content, rule_output_path / test.name)
+
+
+def _process_rules(env_yaml: Dict, output_path: pathlib.Path,
```

Comment:
Code Climate complains about code complexity of this function. Please split it to multiple functions. You can for example extract code related to templated tests to a new function _process_templated_tests which would be consistent with _process_local_tests.